src/language/language_type.hbs.rs

```diff
@@ -396,29 +418,52 @@ impl LanguageType {
         }
 
         let extension = fsutils::get_extension(&entry);
-        let filetype = extension.as_ref()
-            .map(|s| &**s)
-            .or_else(|| get_filetype_from_shebang(&entry));
-
-        if let Some(extension) = filetype {
-            /*
-            if let Some(ref languages) = config.languages {
-                for (language, lang_config) in languages {
-                    if lang_config.extensions.iter().any(|e| e == extension) {
-                        return Some(*language)
-                    }
-                }
-            }
-            */
+        match extension {
+            Some(extension) => LanguageType::from_file_extension(extension.as_str()),
+            None => LanguageType::from_shebang_path(&entry),
+        }
+    }
 
-            match extension {
+    /// Get language from a file extension.
+    ///
+    /// ```no_run
+    /// use tokei::LanguageType;
+    ///
+    /// let rust = LanguageType::from_file_extension("rs");
+    ///
+    /// assert_eq!(rust, Some(LanguageType::Rust));
+    /// ```
+    pub fn from_file_extension(extension: &str) -> Option<Self> {
+        match extension {
+            {{~#each languages}}
+                {{~#if this.extensions}}
+                    {{~#each this.extensions}}
+                        "{{~this}}" {{~#unless @last}} | {{~/unless}}
```

Comment:
You can actually write the match OR `|` preceding your pattern and then it is always allowed even when there is only element. This change was introduced after I wrote this original code. This was designed for exactly this case where it's easier for codegen.
```suggestion
                        | "{{~this}}"
```